src/vivarium_cluster_tools/vipin/log_parser.py

```diff
@@ -0,0 +1,91 @@
+from loguru import logger
+from pathlib import Path
+from random import sample
+import numpy as np
+import pandas as pd
+import json
+import re
+from pandas.io.json import json_normalize
+from typing import Union
+import requests
+
+
+class WorkerLog:
```

Comment:
It's not clear that this class is doing a lot of work for you at this point.  Talked about a `to_df` method in the other PR, but maybe a good thing to do here is to write a docstring and describe what the responsibility of this class is.  

I'd expect a class called `WorkerLog` to be a container representing the contents of an on disk log, but what's here is more of a lazy log parser.

Reply:
WorkerLog was changed to PerformanceLog, but the critique is still the same. I'll add some docstring to help disambiguate what its role is.